pkg/logger/logger.go

```diff
@@ -67,9 +83,11 @@ var currentLogFormat = HUMAN
 
 // Entry defines a log entry
 type Entry struct {
-	Time     string `json:"time"`
-	Severity string `json:"severity"`
-	Message  string `json:"message"`
+	Time     string    `json:"time"`
+	Severity string    `json:"severity"`
+	Code     ErrorCode `json:"code"`
+	Fatal    bool      `json:"fatal"`
```

Comment:
Will we need fatal if we have severity?

Reply:
So log.Fatal will kill the program. We don't want that, we also cannot assume that anything .Error is fatal.

https://github.com/leaktk/scanner/pull/59/files#diff-a7ff0d3ae4aee6ef9557be809cb7b567203efb928f66ef45fee67edbf8dedf3cR131 feels like a bit of a hack, was trying to think of ways to signify that something is scan stopping!
The other idea i had was adding FatalScanError, FatalCloneError. That felt like it would get quite out of control.
Could add a `fatal` parameter but its extra.. 

I figured that all those errors we are sending that are fatal are going to be Errors but not enough to kill the program. Hence the above change. 

Still not 100% sure on this.

Reply:
I was thinking we'd just set the severity to `CRITICAL` and that basically be that the process fell on its face for that request. And basically resource would only have `.Critical(...)` and no `.Fatal(...)` logging on it since we don't want to stop anything.

Thoughts?

Reply:
Seems reasonable. Have CRITICAL, log it out on logger.Error and mark it as Critical

Reply:
logger has a CRITICAL level but no implementation :/